test/typecheck/main.go

```diff
@@ -204,7 +204,7 @@ func (c *collector) verify(plat string) ([]string, error) {
 
 	for _, pkg := range allList {
 		if len(pkg.GoFiles) > 0 {
-			if len(pkg.Errors) > 0 {
+			if len(pkg.Errors) > 0 && (pkg.PkgPath == "main" || strings.Contains(pkg.PkgPath, ".")) {
```

Comment:
Suggestion: maybe this will be more readable:

	hasErrors := len(pkg.Errors) > 0
	isMainOrContainsDot := pkg.PkgPath == "main" || strings.Contains(pkg.PkgPath, ".")
	
	if hasErrors && isMainOrContainsDot {
		errors = append(errors, pkg.Errors...)
	}